pkg/core/codeset_store.go

```diff
@@ -1,46 +1,49 @@
 package fuseml
 
 import (
+	"context"
+
 	"github.com/pkg/errors"
 
-	codeset "github.com/fuseml/fuseml-core/gen/codeset"
-	giteaadmin "github.com/fuseml/fuseml-core/pkg/core/gitea"
+	"github.com/fuseml/fuseml-core/gen/codeset"
 )
 
-type CodesetStore struct {
-	// FIXME this is just internal representation, it should go away
-	items map[string]*codeset.Codeset
+type CodesetStore interface {
+	Find(ctx context.Context, project, name string) (*codeset.Codeset, error)
+	GetAll(ctx context.Context, project, label *string) ([]*codeset.Codeset, error)
+	Add(ctx context.Context, c *codeset.Codeset) (*codeset.Codeset, error)
 }
 
-var (
-	codesetStore = CodesetStore{items: make(map[string]*codeset.Codeset)}
-)
+type GitAdmin interface {
+	PrepareRepo(code *codeset.Codeset) error
+	GetRepos(org, label *string) ([]*codeset.Codeset, error)
+	GetRepo(org, name string) (*codeset.Codeset, error)
+}
 
-func (cs *CodesetStore) FindCodeset(project, name string) (*codeset.Codeset, error) {
+type inMemCodesetStore struct {
```

Comment:
I do not really like this `inMemCodesetStore` ... maybe it's just a name, I am not sure. 
I do not think we plan to have different stores, so it seems unnecessary to me

Reply:
I know what's bothering me about it. That the real store is (currently) git server anyway. So "inMem" is only additional structure, but not the primary one. So it's just confusing to have it there so prominently.

Reply:
You're right, it should say gitCodesetStore. Even better, we can get rid of the in-memory bits, since they're no longer needed with your changes anyway, I checked.